### PR TITLE
Enhancement/41 Podcasts Taxonomy term(s) added in block settings

### LINKED ISSUE #41: Gutenberg: Set taxonomy term(s) in block settings
Right now we are using the default taxonomy panel picker but this would be far better set as a part of adding/editing a podcast block.

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change

* This PR adds the podcasts taxonomy panel picker in the Block settings.
* Please note, that the taxonomy selection would be still connected with the post itself, not the block.

Closes #41 

### Alternate Designs

image

### Possible Drawbacks

* Not a drawback but, the default taxonomy panel under the "Post" tab is still there because if we remove it by making `show_in_rest: true`, we can not replicate it under the settings tab too. If we still need that removed, we may have to add `display: none` to it.

### Verification Process

1. Switch to the PR branch.
2. Run `npm run build`.
3. Create a new/edit existing post.
4. Select the Podcast block.
5. See the podcast taxonomy panel under the block settings.
6. Make sure it works.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

> Added - Podcasts Taxonomy term(s) added in block settings

### Credits

Props @helen @jeffpaul @faisal-alvi

### CODE REVIEW

File: assets/js/edit.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ const { Fragment } = wp.element;
 const { apiFetch } = wp;
 const ALLOWED_MEDIA_TYPES = [ 'audio' ];
 
+// Import taxonomy panel.
+import PostTaxonomies from './taxonomy-panel';
```
Comment (peterwilsoncc) [inline assets/js/edit.js]: Nit: I'd give the component and file the same name: either `PostTaxonomies from './post-taxonomies'` or `TaxonomyPanel from ...`

File: assets/js/taxonomy-panel/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	PostTaxonomies as PostTaxonomiesForm,
+	PostTaxonomiesCheck,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import TaxonomyPanel from './taxonomy-panel';
+
+function PostTaxonomies() {
```
Comment (peterwilsoncc) [inline assets/js/taxonomy-panel/index.js]: Is it possible to simplify this with `<HierarchicalTermSelector slug="podcasting_podcasts" />` similar to how the category selector is shown on the post-publish-check page (but without the messy checking of whether or not to display it)?  

https://github.com/WordPress/gutenberg/blob/5d5e97abd5e082050fdbb88bb1c93f9dbe10a23b/packages/editor/src/components/post-publish-panel/maybe-category-panel.js#L77-L84

**Review by peterwilsoncc — DISMISSED** [review]
This looks good to me but I think it might be possible to simplify the code by using the Gutenberg components.

**Comment by faisal-alvi** [comment]
@peterwilsoncc thanks for the suggestion, I have implemented `HierarchicalTermSelector`. Please let me know if any chagnes are required.

image

File: assets/js/edit.js [hunk 3]
```diff
@@ -20,6 +20,9 @@ const { Fragment } = wp.element;
 const { apiFetch } = wp;
 const ALLOWED_MEDIA_TYPES = [ 'audio' ];
 
+// Import hierarchical term selector.
+import HierarchicalTermSelector from './term-selector/hierarchical-term-selector';
+
```
Comment (peterwilsoncc) [inline assets/js/edit.js]:
```suggestion
/*
 * Import hierarchical term selector.
 *
 * @TODO Import from `@wordpress/editor` once minimum WP version is 6.0.
 */
import HierarchicalTermSelector from './term-selector/hierarchical-term-selector';

```

**Review by peterwilsoncc — DISMISSED** [review]
Thank you, this LGTM with one suggestion to add a todo.

**Review by peterwilsoncc — APPROVED** [review]
Thanks!

**Comment by jeffpaul** [comment]
@faisal-alvi is this good to come out of Draft (and then get merged in)?

**Pull request merged**